core/kernel/delay.c

```diff
@@ -32,6 +32,20 @@
 #include <kernel/misc.h>
 
 #ifdef CFG_CORE_HAS_GENERIC_TIMER
+int timeout_elapsed_us(uint64_t expire)
+{
+	int64_t diff = delay_cnt_read() - expire;
+
+	if (MUL_OVERFLOW(diff, ULL(1000000), &diff)) {
```

Comment:
Is `ULL()` useful here?

Reply:
I don't think it is. I used it to mimic this old implementation.
I'll drop it :).